M/Mesa_EGL/build_tarballs.jl

```diff
@@ -0,0 +1,120 @@
+
+# Fixup paths in Wayland pkg-config files
+sed -i "s?prefix=.*?prefix=${host_prefix}?" "${host_prefix}/lib/pkgconfig/wayland-scanner.pc"
+sed -i "s?prefix=.*?prefix=${host_prefix}?" "${host_prefix}/lib/pkgconfig/wayland-client.pc"
+sed -i "s?prefix=.*?prefix=${host_prefix}?" "${host_prefix}/lib/pkgconfig/wayland-server.pc"
+
+meson -D egl=enabled \
+      -D gles1=enabled \
+      -D gles2=enabled \
+      -D opengl=true \
+      -D platforms=x11,wayland \
+      -D glx=dri \
+      -D c_args="-Wno-implicit-function-declaration" \
+      -D cpp_rtti=false \
+      ../ \
+      --cross-file="${MESON_TARGET_TOOLCHAIN}" \
+      --cross-file=llvm-cross.ini
+ninja -j${nproc}
+ninja install
+
+# taken from https://metadata.ftp-master.debian.org/changelogs//main/m/mesa/mesa_20.3.5-1_copyright
+install_license ../../copyright
+"""
+
+# TODO: Hack to ensure we load the right drivers
+init_block = """
+ENV["LIBGL_DRIVERS_PATH"] = joinpath(artifact_dir, "lib", "dri")
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11"),
+]
+
+# The products that we will ensure are always built
+products = Product[
+    LibraryProduct("libEGL", :libEGL),
+    LibraryProduct("libGLESv1_CM", :libGLESv1_CM),
+    LibraryProduct("libGLESv2", :libGLESv2),
+    LibraryProduct("libGL", :libGL),
+    LibraryProduct("libvulkan_intel", :libvulkan_intel),
+    LibraryProduct("libvulkan_lvp", :libvulkan_lvp),
+    LibraryProduct("libvulkan_radeon", :libvulkan_radeon),
+    LibraryProduct("libxatracker", :libxatracker),
+    LibraryProduct("libgbm", :libgbm),
+    LibraryProduct("libglapi", :libglapi),
+
+    # Drivers
+    LibraryProduct("crocus_dri", :crocus_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("iris_dri", :iris_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("libgallium_dri", :libgallium_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("r300_dri", :r300_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("radeonsi_dri", :radeonsi_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("virtio_gpu_dri", :virtio_gpu_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("i915_dri", :i915_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("kms_swrast_dri", :kms_swrast_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("nouveau_dri", :nouveau_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("r600_dri", :r600_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("swrast_dri", :swrast_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("vmwgfx_dri", :vmwgfx_dri, ["lib/dri"]; dont_dlopen=true),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    HostBuildDependency("Wayland_jll"),
+    Dependency("libLLVM_jll"; compat="11.0.0"),
```

Comment:
So this will work only in julia v1.6?

Reply:
@vchuravy what's the kosher way of dealing with this?  Use `libLLVM_jll` as dependency but not specify a compat?  Use `libLLVM_jll` as a `BuildDependency` (libllvm will be anyway available at runtime)?  Or perhaps follow [`LLVMExtra`](https://github.com/JuliaPackaging/Yggdrasil/blob/5dac293be9a0af939f61f6298eec08c09207a8ab/L/LLVMExtra/build_tarballs.jl) now that we have that mechanism in place?

Reply:
Ok, seeing that this links directly to `libLLVM-11jl.so` I think the the LLVMExtra's approach is the way to go.  I'm actually surprised it worked for you with other Julia versions with different LLVM :thinking:

Reply:
Oh, no, it didn't :smile: I had to switch which libLLVM I imported.